service/resource/certificate/update.go

```diff
@@ -42,6 +42,11 @@ func (r *Resource) ProcessUpdateState(ctx context.Context, obj, updateState inte
 		return microerror.Mask(err)
 	}
 
+	// In case the update state is nil, don't process at all.
+	if updateCertificateFiles == nil {
+		return nil
+	}
```

Comment:
I guess it's better to check `len(updateCertificateFiles) == 0`

Reply:
This could be the case where there should be no certificates. nil is that the update should not be processed.

Reply:
This assumption/approach is also used in other operators so far. The zero value should always be considered empty implying to do nothing. An empty initialized list might mean to update by purging every state.